src/Filesystem/Folder.php

```diff
@@ -416,21 +416,26 @@ public function inCakePath($path = '')
     }
 
     /**
-     * Returns true if the File is in given path.
+     * Returns true if the Folder is in the given path.
      *
-     * @param string $path The path to check that the current pwd() resides with in.
-     * @param bool $reverse Reverse the search, check that pwd() resides within $path.
+     * @param string $path The absolute path to check that the current `pwd()` resides within. If omitted
+     *  (or empty), the current top level directory is assumed.
+     * @param bool $reverse Reverse the search, check if the given `$path` resides within the current `pwd()`.
      * @return bool
      */
     public function inPath($path = '', $reverse = false)
     {
+        if (empty($path)) {
```

Comment:
If it never worked and emitted a warning, adding a formal exception seems ok to me.